1_fetch/src/get_nhdplusv2.R

```diff
@@ -0,0 +1,18 @@
+get_nhdv2_flowlines <- function(huc8){
+  #' 
+  #' @description Function to download NHDPlusV2 flowlines within individual HUC8 watersheds
+  #'
+  #' @param huc8 string or character vector containing the HUC8 watersheds of interest
+  #'
+  #' @value An sf object containing NHDPlusV2 flowlines
+  
+  huc8_basin <- suppressMessages(nhdplusTools::get_huc8(id=huc8))
+  huc8_bbox <- sf::st_as_sfc(st_bbox(huc8_basin))
+  
+  # Download flowlines within the bounding box of the HUC8 basin
+  # Use the bounding box rather than the basin boundary to make sure we're capturing all of the reaches
```

Comment:
Interesting. Did you see cases where some reaches were being missed with the huc polygons (as opposed to the BB)?

Reply:
I opted to use bounding boxes because I noticed the most downstream PRMS reaches don't get returned if I retrieve NHD by the HUC8basins:

![compare](https://user-images.githubusercontent.com/8785034/147283647-09a90f10-6362-4236-baee-171e763c1b14.png)

I'm not sure if this really matters for our purposes since we're focused on the upstream segments, but I more or less thought the bbox approach was more conservative in terms of the NHD reaches that get returned.

Reply:
It looks like I didn't update the legend labels in the above figure, but the panel on the right should read "NHD by bbox"